### ISSUE TITLE
Restarting tween right after initialization makes it stuck

### ISSUE THREAD

**Issue by Antriel** [photonstorm/phaser#3651]
Example code:
```js
var tween = this.tweens.add({
    targets: { v:0 },
    v: 100,
    onComplete: function() { console.log("never fires"); }
});
tween.restart();
```

This causes the tween to have a `TWEEN_CONST.ACTIVE` state, but it's still in `TweenManager`'s `_add` because it checks for state of `TWEEN_CONST.PENDING_ADD`.
A possible fix would be not restarting tween that has such state or changing the check in `TweenManager`.

**Comment by zacharysarette** [comment]
I just created a PR for this here #3680 .

It does look like it checks the state in the restart function to see if it is TWEEN_CONST.REMOVED but doesn't check if it is TWEEN_CONST.PENDING_ADD.

I added it to the if statement and tested it. It works. 

We'll see if it gets pulled in or not. If someone has a better idea on how to fix the issue I would love to see how the solution. Any advice would be appreciated since I am new to contributing to Phaser.js. Cheers!
